File: src/models/PromotionalCode.ts
import WritableDbItem from "./WritableDbItem";

/** A promotional code and the client that redeemed it, when applicable. */
export default interface PromotionalCode extends WritableDbItem {
  type: CodeType;
  redeemerId: string | null;
}

export enum CodeType {
  UNKNOWN,
  CITIZEN_CON_2954,
}

File: src-dotnet/models/PromotionalCode.cs
namespace MedrunnerApiClient.Models;

/// <summary>
/// Represents a promotional code.
/// </summary>
public class PromotionalCode : WritableDbItem
{
    /// <summary>
    /// The ID of the redeemer.
    /// </summary>
    public string RedeemerId { get; set; }

    /// <summary>
    /// The type of code.
    /// </summary>
    public CodeType Type { get; set; }
}

/// <summary>
/// The type of promotional code.
/// </summary>
public enum CodeType
{
    Unknown,
    CitizenCon2954
}
